add DragonFlyBSD support to modules/system/user.py

jontow opened the pull request:
##### SUMMARY
Duplicate existing FreeBsdUser class as DragonFlyUser, correctly setting platform name.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
system/user

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0 (devel 97b51f6187) last updated 2018/01/21 18:28:36 (GMT -400)
  config file = None
  configured module search path = ['/Users/jontow/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jontow/GIT-external/ansible/lib/ansible
  executable location = /Users/jontow/GIT-external/ansible/bin/ansible
  python version = 3.6.4 (default, Jan  4 2018, 20:49:52) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
This is required on DragonFly, or it fails with an error like 'Failed to find required executable usermod' (DragonFly uses 'pw', like FreeBSD).
Instead of attempting something fancy like loading the FreeBsdUser class and overriding just the platform, I chose to follow existing convention and create a new class, since it is technically a separate platform that can diverge over time.
To trigger this bug, simply define a user that gets applied on a DragonFlyBSD host, like so:
```
---
- user:
    name: jontow
    comment: "jontow"
    groups: "{{ admin_groups }}"
    append: yes
    shell: "{{ zsh_path }}"
    update_password: on_create
    password: '...'
```

<!--- Paste verbatim command output below, e.g. before and after your change -->
Before:
```
TASK [common : user] *********************************************************************************************************************************************
task path: /Users/jontow/GIT-je/ansible/roles/common/tasks/users/jontow.yml:2
fatal: [dbuild.je]: FAILED! => {"attempts": 1, "changed": false, "failed": true, "msg": "Failed to find required executable usermod in paths: /home/jontow/.rbenv/bin:/home/jontow/.cargo/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/usr/pkg/sbin:/usr/pkg/bin:/home/jontow/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/bin:/usr/local/sbin:/usr/pkg/bin:/usr/X11R6/bin:/home/argus/bin:/home/argus/sbin:/home/jontow/bin"}
```

After:
```
TASK [common : user] *********************************************************************************************************************************************
task path: /Users/jontow/GIT-je/ansible/roles/common/tasks/users/jontow.yml:2
ok: [dbuild.je] => {"append": true, "attempts": 1, "changed": false, "comment": "Jonathan Towne", "failed": false, "group": 1000, "groups": "wheel", "home": "/home/jontow", "move_home": false, "name": "jontow", "password": "NOT_LOGGING_PASSWORD", "shell": "/usr/local/bin/zsh", "state": "present", "uid": 1000}
```

Akasurde committed:
Remove deprecated get_exception API

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Adds the oVirt High Performance (high_performance) VM type.

agaffney committed:
Add support for defining index var for task loops

Commit:
add support for configuring smtpport in callback_mail

Akasurde committed:
VMware: vmware_guest template find (#35088)

This fix adds logic to find unique virtual machine template using name.

Fixes: #26669

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Fix junos_facts integration test failure (#35469)

Hostname is not always configured on remote host
Instead add a assert statement to check if netconf
is configured for junos_facts.

orthanc committed:
New aws_ses_identity module to manage AWS Simple Email Service Identity (#31140)

* Add aws_ses_identity module

* Update CI alias, add BotoCoreError exception handling.

* Add SES and SNS permissions to hacking/aws_config to run aws_ses_identity integration tests

gundalow committed:
Infoblox should be run locally (#35390)

* Infoblox should be run locally

* use connection: local

alikins committed:
intg tests for 'any_errors_fatal' (#25813)

* start adding intg tests for any_errors_fatal

* updates

* show captured output

flip conditional for on_includes

* cleanup/ci poke

* fix typos

* Add 'EXPECTED FAILURE' to tasks names for expected failures

s-hertel committed:
[cloud] ec2_eni: Add a purge option for secondary ip addresses - fixes #26575 (#26596)

* add purge option for secondary ip addresses and fix changed to reflect when addresses are modified

* Improve option documentation and make the secondary_private_ip_addresses option required if purging secondary private ip addresses.

* ec2_eni: update version_added to 2.5 for purge_secondary_private_ip_addresses option.

abadger committed:
Clarify lookup plugin TypeError recovery

Change the code and comments around recovering from a TypeError in
handling the return value from lookup plugins to make it clearer what
we're recovering from.

jborean93 committed:
Added code-smell check to verify filenames are allowed with Windows (#35436)

mattclay committed:
Fix win_regedit module and tests. (#35478)

s-hertel committed:
[cloud] new module - AWS Direct Connect virtual interface (#33211)

* Add an AWS Direct Connect virtual interface module

Fix DirectConnectError exception wrapper to be compatible with fail_json_aws()

* Fix documentation and handle BotoCoreErrors as well

* Add the aws and ec2 documentation fragments

ryansb committed:
Add AWS Direct Connect VIF (aws_direct_connect_virtual_interface) to CHANGELOG

Akasurde committed:
nmcli: Allow setting ipv4.dhcp-client-id parameter (#34268)

Signed-off-by: Orion Poplawski <orion@cora.nwra.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Use static password for win_owner test. (#35482)

This will provide consistent results across test runs and
should avoid previously seen issues with password complexity
failures.

dagwieers committed:
Add win_product_facts

viper233 committed:
Elasticbeanstalk app (#31079)

* New AWS elasticbeanstalk app module

* Integration tests for elasticbeanstalk_app

mattclay committed:
Test Windows 2012+ using Azure by default.

mattclay committed:
Temporarily disable `influxdb_user` test.

The test interferes with the `uri` test.

privateip committed:
fixes error in dict_merge when merging list value (#35461)

brianteeman committed:
english corrections (#35307)

chipgwyn committed:
bug-35431 fixed removing encryption from user on subsequent runs of the task (#35433)

mkrizek committed:
Fix triggering parent's always with run_once set (#35464)

gundalow committed:
Module deprecation: docs, scheme and tests (#34100)

Enforce module deprecation.
After module has reached the end of it's deprecation cycle we will replace it with a docs stub.

* Replace deprecated modules with docs-only sub
* Use of deprecated past deprecation cycle gives meaningful message (see examples below)
* Enforce documentation.deprecation dict via `schema.py`
* Update `ansible-doc` and web docs to display documentation.deprecation
* Document that structure in `dev_guide`
* Ensure that all modules starting with `_` have a `deprecation:` block
* Ensure `deprecation:` block is only used on modules that start with `_`
* `removed_in` A string which represents when this module needs **deleting**
* CHANGELOG.md and porting_guide_2.5.rst list removed modules as well as alternatives
* CHANGELOG.md links to porting guide index

To ensure that meaningful messages are given to the user if they try to use a module at the end of it's deprecation cycle we enforce the module to contain:
```python
if __name__ == '__main__':
    removed_module()
```

sivel committed:
Add porting guide note about include inheritance change (#35199)

* Add porting guide note about include inheritance change. Fixes #35096

* Also make note of using import_* when possible

* Edited for clarity.

dsoper2 committed:
Add support for primary and secondary redundancy types.  Require cdn name for user-defined names. (#35505)

Qalthos committed:
Fix ios_facts return values (#35239)

* Revert model and serialnum to older version

* Add stacked versions of model and serialnum as separate facts

* Add unit test to check stacked output

* Alter model regex to address #34768

maxamillion committed:
galaxy remove incorrect check for role path before extraction (#35259)

* galaxy remove incorrect check for role path before extraction

Fixes #35217

Currently lib/ansible/galaxy/role.py checks roles_path and will
ignore paths that don't currently exist which it should not because
the path will attempt to be created by the role extraction process
(which is valid and expected behavior) and if unable to write at the
time of role extraction into the role_path, that error will be
handled properly at that point in the process.

Signed-off-by: Adam Miller <admiller@redhat.com>

* add test case for galaxy install with rolefile and rolepath

Verify we don't regress GitHub Issue #35217

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix the integration test

Signed-off-by: Adam Miller <admiller@redhat.com>

* double quote for shellcheck glob warning

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix galaxy install tests to not need remote resources

Signed-off-by: Adam Miller <admiller@redhat.com>

* make shellcheck sanity tests happy

Signed-off-by: Adam Miller <admiller@redhat.com>

* prep git global conf for running the galaxy install tests in docker

Signed-off-by: Adam Miller <admiller@redhat.com>

* move ansible-galaxy tests into their own target, make git conf non-global

Signed-off-by: Adam Miller <admiller@redhat.com>

* fix up tests based on feedback

Signed-off-by: Adam Miller <admiller@redhat.com>

* remove extra newline from aliases file

Signed-off-by: Adam Miller <admiller@redhat.com>

alikins committed:
Fix virt role detection for kvm if no other match (#35403)

This fixes an indention bug introduced in
45a9f967749ec68e2077fe1d1d32dd37660ab376
that was causing the virtualization_role to be set to 'host'
for many cases incorrectly.

Fixes https://github.com/ansible/ansible/pull/21621
(the issue mentioned in the pr)

privateip committed:
exposes rpc method to plugin (#35507)

The rpc method should be exposed to the modules over the connection rpc
mechanism.  This fix will expose the rpc method to call generalized rpc
endpoints on remote devices

jonmeran committed:
Initial support for AWS Batch Compute Environment (#26699)

* Add AWS Batch Compute Environment module

* Move common code for AWS batch modules in module_utils/aws/batch

ryansb committed:
Add aws_batch_compute_environment to CHANGELOG

ganeshrn committed:
Fix junos integration test failures (#35508)

* More integration testcase fixes for junos_facts

sivel committed:
Guard against assignments that are not ast.Name in docs parsing (#35513)

hatuomin committed:
Netact cm command2 (#34873)

s-hertel committed:
[cloud] iam_role: allow retention of existing policies (#28513)

* module_iam_role: add_remove_policies_option

* Update 24080 with requested changes.

* Fix whitespace and rename the option remove_unlisted_policies to purge_policies.

Update version_added

* Fix exception handling in iam_role module

* Fix line length

s-hertel committed:
Add new AWS modules aws_elasticbeanstalk_app and aws_ses_identity (#35518)

s-hertel committed:
elasticache: default zone to None - fixes #31779 (#31785)

* elasticache: default zone to None.

* Only add PreferredAvailabilityZone to kwargs if self.zone is not None

sivel committed:
Replace LooseVersion comparison with int/float comparison. Fixes #35435 (#35519)

clementtrebuchet committed:
Fixes #35468: ADD max_results to the nios api (#35473)

* Fixes #35468: ADD max_results to the nios api

* Fixes #35468: fix nios unittest

* Fixes #35468: document the module docs fragments for nios

jonmeran committed:
[cloud] new module aws_batch_job_queue: support for AWS Batch Job Queues (#26701)

1. Removed redundant default values in argument_spec
2. Enclosed non standard python libraries in Try Except ImportError block
3. Added region as a mandatory element in argument_spec
4. Updated the typo in api_params while accessing job queue. Changed it from job_queue to jobQueue
5. Updated the set_api_params() method to use snake_dict_to_camel_dict()
6. Updated the ANSIBLE_METADATA version from 1.0 to 1.1
7. Updated the version_added in Documentation from 2.4 to 2.5
8. Initialized response value to None
9. Updated the RETUN documentation
10. Updated the return from manage_state() method to not embed batch_job_queue_action in ansible_facts
11. Updated the module.exit_json()  in main() to convert any keys back to snake_dict format using camel_dict_to_snake_dict to maintain consistency throughout the module
12. Changed the filename from batch_job_queue.py to aws_batch_job_queue.py

ryansb committed:
New module: aws_batch_job_definition

ryansb committed:
Add aws_batch_job_queue and aws_batch_job_definition to CHANGELOG

felixfontein committed:
Changing letsencrypt module to use ACME v2 protocol (update) (#35283)

* Adding warnings.

* Forgot choices for acme_version in code.

* Removed 'errors' support again.

* For DNS challenges, also return complete record to simplify wildcard generation (see #35283).

* Also returning order URI and account URI. This is mainly for debugging purposes.

* Adding more clear separation between the two calls. Avoids problems where code during second call thinks it is during first call and doesn't stop with error, but doesn't obtain a new certificate either.

* Added validate_certs parameter.

* Actively discouraging from setting validate_certs to false.

* Fixing DNS challenge example.

* Adding new output challenge_data_dns, which simplifies DNS challenges.

Commit:
Add jenkins_job_facts module (#35346)

* Add jenkins_job_facts module

* Fix docs

* Fix connection error handling

* Add support for filtering by color

* Miscellaneous code cleanup

* Spell out "underscore" in comment

* Update required state in docs

abadger committed:
Test behaviour of loop keyword plus lookup plugins

We introduced the new loop keyword as a replacement for with without
adding tests that it behaved as we expected.  This test asserts that
behaviour.

Incidentally, it also shows how to use parameters with lookups and loops
now.

jborean93 committed:
change .encode to to_bytes (#35522)

ignore backup dir in test integration targets folder

credmond-git committed:
open_iscsi fails to login when discovery returns multiple portals #29309 (#35262)

tmshn committed:
timezone module: Enhanced update command for Debian/Ubuntu (#34390)

* timezone module: Symlink to /etc/localtime before dpkg-reconfigure in Debian/Ubuntu

* timezone module: Symlink rather than hard link

mattclay committed:
Add pslint sanity test settings.

Globally ignore rule: PSUseShouldProcessForStateChangingFunctions

kyliel committed:
Update guide_azure.rst to include note about running ansible from within azure (#35441)

* Update guide_azure.rst

Add one more option to run Ansible in Azure

* Minor edit

Commit:
fix nxos_pim_interface issues (#35405)

* fix nxos_pim_interface issues

* add absent test for pim_interface

ganeshrn committed:
Fix double prompt issue in network_cli (#35439)

Fixes #33993

In current implementation while trying to fetch the
current cli prompt terminal plugin executes `\r` on remote host.

This causes if command execution results in a prompt for
example executing `enable` command results in `Password:` prompt
there by executing `\r` on password prompt which is not intended.

To fix it while fetching current prompt from terminal plugin use `get_prompt()`
api in network_cli that returns the latest matched prompt instead of
unnecessary executing `\r` on remote host.

felixfontein committed:
letsencrypt: fix account key detection for keys created by openssl_privatekey (#35534)

* Fixing key detection if key was generated with openssl_privatekey.

* Fixing error formatting.

Commit:
Changing order of load and loadfp (#34793)

* Changing order of load and loadfp

* Removing exception as per Ganesh Review comments

gundalow committed:
Network Doc status (#35475)

Plus remove trailing whitespace

Commit:
Fixing Bug for Embedded Switches in Parsing data (#35541)

machacekondra committed:
ovirt_auth: Fix checking of auth paramters (#35512)

KKoukiou committed:
ovirt_storage_domains: docs: override_luns parameter is applicable also to fcp (#35506)

Commit:
[cloud] Improve ipv6 and EC2 classic support in ec2_group integration tests (#32976)

* ec2_group: fix ipv6 tests to use an explicit VPC

* otherwise would fail on old AWS accounts supporting EC2-classic

* ec2_group: fix tests to use an explicit VPC

* Only run some tests if there is a default vpc associated with the account

ganeshrn committed:
Handle multiple sub prompts for network_cli connection (#35361)

* Handle multiple sub prompts for network_cli connection

Fixes #35349

*  Check if the same prompt is repeated in consecutive window
   if it is repeated it indicates there is problem with answer
   provided
*  In that case report error to user

* Fix CI failure

* Fixes #35349

*  Add prompt_retry count to control max number of times
   to expect the same prompt before it error's out

*  Make required changes in ios and eos terminal plugin to handle
   wrong enable password correctly and return proper error
   message to user.

*  Check if the same prompt is repeated in consecutive window
   if it is repeated it indicates there is the problem with an answer
   provided

*  In that case report error to user

eikef committed:
New module: keycloak_clienttemplate (#33419)

* keycloak_clienttemplate

* BOTMETA maintainership for identity/keycloak namespace

* fix superfluous blank line

* catch ValueError when trying to decode JSON

* further documentation for protocol mappers and some checks

* whitespace fixes, YAML fixes

* remove state: dump, update argument_spec and documentation with suboptions

* add documentation for realm option

* document aliases for auth_keycloak_url, auth_username, and auth_password (i.e. url, username, and password)

* remove bearer_only, consent_required, standard_flow_enabled, implicit_flow_enabled, direct_access_grants_enabled, service_accounts_enabled, public_client, and frontchannel_logout from module options.

kiorky committed:
FIX: multiple nested tasks include levels (#35165)

* Test for include_tasks & include_role bug

Related to ansible/ansible:#21890

* Fix nested include_tasks called from role

This fixes the path of included files when you want
to call include_task inside a role's task file and this
role is itself called from multiple level of playbook include_tasks

Related to ansible/ansible:#21890
This fixes ansible/ansible:#35109

clementtrebuchet committed:
Contrib infoblox exattrs Fixes #35409 (#35410)

* *FIX: re-create the exattrs dict into the hostvars[name] dict

* *FIX: remove unwanted files

* Fixes #35409: put an str instead of variable

ericwb committed:
ios_facts: consistently use type of string not str (#35104)

The return documentation for ios_facts used an inconsistent mix
of str and string to denote a string data type.

This patch modifies all to be "string".

Commit:
Changing ENOS cliconf plugin license from BSD to GPL (#34889)

* Changing BSD to GPL

gundalow committed:
Remove network-test-runner (#35556)

Not been used for a while, replaced by DCI

sivel committed:
Add configurable blacklist filtering for python logger

Akasurde committed:
VMware: Add basic vmware module related guide (#35128)

* VMware: Add basic vmware module related guide

This fix adds page tries to consolidate various information related
to vmware modules and debugging related information.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

* Copy edits

* Added module reference

* Edits

* Fixing invalid characters

Commit:
creates structure for network docs (#35514)

s-hertel committed:
[cloud] Log more exception info to fix sts_assume_role integration tests (#35570)

bgurney-rh committed:
New Ansible module for Virtual Data Optimizer (#32267)

Add an Ansible module for VDO (Virtual Data Optimizer), a
device-mapper module that provides deduplication, compression,
and thin provisioning.  This module provides the ability to create,
remove, and modify parameters on storage volumes created by VDO.

abadger committed:
Update the release doc with ansible-2.4.3

abadger committed:
Update version in the versions.yml file

(cherry picked from commit c8941cb10301d4f4519b2c18cf4d4fd6d3ba0f75)

Commit:
Reflect new process for network integration tests (#35577)

With `ansible-test network-integration` we don't need the top level yaml files
except NXOS and Dell (for now)

abadger committed:
Correct message about what files to double check for release

abadger committed:
Add 2.4.3 to releases.txt

dagwieers committed:
aci_aaa_user: Manage AAA users on ACI fabrics (#35543)

* aci_aaa_user: Manage AAA users on ACI fabrics

* Fix module documentation

* Ensure we allow to not set accountStatus and expires

* Added aaa_password_lifetime and aaa_password_update_required support

* Improvements to integration tests in light of issue 35544

* Fix ACI ISO 8601 formatted string

* Add HAS_DATEUTIL

mattclay committed:
Remove EOL openSUSE Leap 42.2 from CI.

mattclay committed:
Fix pip integration test.

Commit:
fixing mysql and postgresql test failures (#35587)

AbbyRH committed:
added force option to zabbix_hostmacro to allow control of overwriting  (#35516)

* added force option to hostmacro to allow control of overwriting existing macro

* wording changes as suggested by eikef in PR 35516

* further wording changes from eikef

* added 'version_added: 2.5' to the documenation for the force option

kedarkekan committed:
* adds commit replace with config file for iosxr (#35564)

* * adds commit replace with config file for iosxr
* fixes dci failure in iosxr_logging

* * review comment changes

acrewdson committed:
Removing temporary files when not proxing inside urls utils (#35583)

rbywater committed:
Add ability to select to prefer IPv4 addresses for ansible_ssh_host (#35584)

Currently Cloudforms can return a mix of IPv4 and IPv6 addresses in the
ipaddresses field and this mix comes in a "random" order (that is the
first entry may be IPv4 sometimes but IPv6 other times). If you wish to
always use IPv4 for the ansible_ssh_host value then this is problematic.

This change adds a new prefer_ipv4 flag which will look for the first
IPv4 address in the ipaddresses list and uses that instead of just the
first entry.

ganeshrn committed:
Check if data_xml object in present in NCElement object (#35604)

If ncclient return NCElement object call data_xml() on response
else call xml()

wilvk committed:
New module: manage Amazon CloudFront origin access identities (cloud/amazon/cloudfront_origin_access_identity) (#35540)

mattclay committed:
Move import sanity test files into own directory. (#35593)

ar7z1 committed:
Integration tests for rabbitmq_user module (#35421)

pilou- committed:
yum_repository: Python 3.7 error have been fixed

See 232dc7110ca1fcaa0147bcce4773845220766869

s-hertel committed:
Fix idempotence for deleting ElasticBeanstalk applications (#35614)

jcea committed:
Missing facts in SunOS module

abadger committed:
Update code smell blacklists (#35621)

Update blacklist for get_exception, boilerplate, and no-wildcard.

f5, lenovo, and junos have updated their code.

samdoran committed:
Fix lineinfile to insert line when pattern exists elsewhere in the file. (#33393)

* Insert lines before or after when the regexp exists elsewhere in the file

* Correct filter syntax in lineinfile integration test

* Use multi-line YAML syntax on lineinfile tests

Unify indentation

* Add lineinfile tests for same line matched to different regexps

* Remove debug statement from test

willthames committed:
Add new aws_waf_condition module (#33110)

privateip committed:
remove restart_if_needed argument (#35626)

* remove restart_if_needed argument

* fix sanity check error

abadger committed:
Fix packaging/os modules for wildcard imports and get_exception

jmighion committed:
Fixing name from aruba to ios. (#35630)

balonik committed:
Make vmware_guest always get a resource pool (#34995)

jcsmith committed:
Update developing inventory script docs. (#35639)

* Update developing inventory script docs.

* Copy edit.

thomwiggers committed:
Include new module openssl_dhparam in changelog (#32620) (#35650)

dagwieers committed:
Simplify auth logic (#35647)

mcavedon committed:
Update playbooks_best_practices.rst (#35627)

* Update playbooks_best_practices.rst

If deploying to the first 10 hosts in Boston it should be "-limit boston[0:9]" and the next 10 should be "-limit boston[10:19]". The doc as written would skip the first host (boston[0]) and try to deploy to a host that doesn't exist (boston[20])

* Typo fix

ivanbojer committed:
remove panos_security_policy (#34433)

* remove panos_security_policy

* added depreciation notice

* - deprecation notice

* properly document deprecated module

* merge changes

* merge changes

* parser expects srt for deprecations attributes

ivanbojer committed:
new module panos_query_rules (#35203)

* new module

lrivallain committed:
vmware: Add new module 'vmware_datastore_host' (#35004)

* vmware: Add new module 'vmware_datastore_host'

Signed-off-by: Ludovic Rivallain <ludovic.rivallain@gmail.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

resmo committed:
doc: fix loop examples (#34991)

trishnaguha committed:
change testfiles for nxos_file_copy (#35641)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

jctanner committed:
Fix loop/with typo in loops doc (#35652)

s-hertel committed:
Make an aws_credentials documentation fragment for plugins using environment vars (#35578)

Akasurde committed:
VMware: Deprecate vsphere_guest module (#35648)

This change deprecates vsphere_guest in favor of vmware_guest and other
related modules.

The major reasons behind deprecation are -

- Pysphere - Unofficial Python bindings of vCenter deprecated in the year 2013.
- VMware provides official Python bindings for vCenter, which is used in vmware_guest.
- vcsim - simulator used in integration testing of vmware module does not support PySphere
  APIs, which makes it more difficult to test vsphere_guest.

Please see [deprecation plan](https://github.com/ansible/community/wiki/VMware%3A-vsphere_guest_deprecation)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Qalthos committed:
ios test changes (#35510)

* Fix over-byte

* Update ios tests to call `provider`

To continue to support testing `connection: local`

* Fix command dict handling in ios_user

* Clean up unit tests, too

Qalthos committed:
Fix paramiko connections to hosts with uppercase characters (#35555)

trishnaguha committed:
nxos_vlan ValueError fix (#35559)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

mkrizek committed:
Add any_errors_fatal in nested failed block test (#35499)

mkrizek committed:
Use value of delegated transport for var lookup (#35658)

mtnbikenc committed:
Update BOTMETA for VMware maintainers (#35609)

maxamillion committed:
make v2_playbook_on_notify method signatures consistent

This was causing an exception in the TaskQueueManager when a third
party handler plugin was processed that inherited or explicitly
called the callback method from super because the method signature
was incorrect in callback/__init__ and it processed the arguments as
incorrect data types as a side effect.

Signed-off-by: Adam Miller <admiller@redhat.com>

pilou- committed:
Check that AnsibleUndefinedVariable doesn't occur when an unused variable references an undefined variable (#35571)

* Check that AnsibleUndefinedVariable doesn't occur

* AnsibleUndefinedVariable exc.: don't modify type

dagwieers committed:
ACI: Implement aci.boolean() to return an ACI boolean (#35610)

* ACI: Implement aci.boolean() to return an ACI boolean

A boolean value in ACI is not always standardized to yes/no.
Sometimes we have active/inactive, or enabled/disabled
Whereas the interface we want is a true YAML boolean.

We did not modify enabled/disabled values at this time.
I first want to determine if this implementation is acceptable.

* Support enabled/disabled as well, with deprecation messages

* Fix typo

* Fix PEP8 issue

* Ensure the aci object exists before using it

* Add comment to ensure this gets fixed in v2.9

* Fix typo

jcsmith committed:
Update some examples to use proper YAML syntax. (#35640)

* Update some examples to use proper YAML syntax.

* Make the requested changes so this builds properly.

Also fix emphasis line numbers to match what was being emphasized before
the change.

dagwieers committed:
aci_epg: Add support for Preferred Group Membership (#35265)

* aci_epg: Add support for Preferred Group Membership

* ACI defaults to 'no'

* Add 'version_added'

* Make use of aci.boolean()

wilvk committed:
[cloud][aws] New module: cloudfront_invalidation (#35601)

* Remove redundant default for target_paths

* Make invalidation requests idempotent

* Remove redundant exception handling around boto3_conn

* Add cloudfront_invalidation to the changelog

s-hertel committed:
Fix syntax error so route table tests always try to clean up (#35663)

* Always try to clean up after route table tests

* Wait for a new VPC a chance to become available before attempting to get its current state

newswangerd committed:
Add vyos smoke integration tests. (#35456)

* added preliminary vyos smoke tests

* added comments, determined current tests are sufficient

* add tests for check mode in module_utils.network.vyos

* removed vyos_smoke from shippable

* added comment explaining why AWS/Shippable tests won't work

willthames committed:
[cloud][aws] New module: aws_waf_rule module (#33124)

Add a new module for managing AWS WAF rules

Preceded by aws_waf_condition and to be succeeded by aws_waf_web_acl

ryansb committed:
Add aws_waf_rule to CHANGELOG

mattclay committed:
Disable failing aws_waf_web_acl tests.

dagwieers committed:
ACI: Change RETURN output as discussed (#35617)

* ACI: Change result output as discussed

* Update all modules to use new aci.exit_json()

* Update output_level spec and docs

* Fix integration tests

* Small PEP8 fix

* Asorted fixes to tests and aci_rest

* More test fixes and support for ANSIBLE_DEBUG

* Fix another PEP8 issues

* Move response handling inside ACI module

* Reform of ACI error handling and error output

* Diff multiline json output

* Fix a few more tests

* Revert aci_bd tests

* Small correction

* UI change: existing->current, original->previous

* UI change: config->sent

* Update all modules with RETURN values

* Fix a few more tests

* Improve docstring and add 'raw' return value

* Fix thinko

* Fix sanity/pep8 issues

* Rewrite unit tests to comply with new design

brunocalogero committed:
added aci_fabric_node module (#35586)

brunocalogero committed:
added aci_static_binding_to_epg module (#35581)

* added aci_static_binding_to_epg module

* fixing pep8 issues

* fixed documentation yaml ':' error

* fix aliases error

* small documentation fixes suggested in PR

mikewiebe committed:
Add minor conditional checks to nxos_bgp sanity test (#35476)

* Add minor conditional checks to nxos_bgp sanity test

* Make ansibot happy

jcsmith committed:
Update example syntax (#35670)

* Update examples in guide_rolling_upgrade.rst.

* Update examples playbooks_best_practices.rst.

* Update examples in playbooks_checkmode.rst.

jcsmith committed:
Update example syntax in playbooks_conditionals.rst (#35673)

jcsmith committed:
Update example syntax in playbooks_delegation.rst. (#35674)

jcsmith committed:
Update example syntax in playbooks_error_handling.rst. (#35675)

jcsmith committed:
Update example syntax in playbooks_filters.rst. (#35676)

jcsmith committed:
Update example syntax in playbooks_intro.rst. (#35677)

jcsmith committed:
Update example syntax in playbooks_lookups.rst. (#35678)

jcsmith committed:
Update example syntax in playbooks_loops.rst. (#35679)

jcsmith committed:
Update example syntax in playbooks_tags.rst. (#35680)

ivanbojer committed:
new module that allows for adding DynamicAddressGroup tags to the firewall (#35201)

* new module that allows for adding dag tags

ivanbojer committed:
removed panos_nat_policy (#34432)

* removed panos_nat_policy

* merge

* removed extra line

resmo committed:
cs_vpc: fix network_domain param ignored (#35690)

prutseltje committed:
Digitalocean public private ip (#27269)

* When using private networking the ip_address key contains the private ip.
Add the following info to module output when available:
  - private_ipv4_address
  - public_ipv4_address
  - private_ipv6_address
  - public_ipv6_address

Akasurde committed:
VMware: Provide verbose message about non-existent VM (#35682)

This fix adds a verbose message about non-existent VM when specified
with operation, due to idempotency we can not detect correct state.

Fixes: #27384

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Refactor vmware_vmkernel module (#35531)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

resmo committed:
cs_network: implement support acl (#35706)

Commit:
fix nxos_ospf_vrf issues (#35616)

mikewiebe committed:
Fix nxos_bgp tests and prepare_nxos_tests refactor (#35694)

paulRbr committed:
contrib(inventory): fix Scaleway inventory used w/env variable token (#35701)

In order to authenticate with the Scaleway API for the contrib dynamic
inventory Scaleway the environment variable SCALEWAY_TOKEN was read
only if a value was set in the configuration .ini file.

This PR fixes that allowing to read only the environment variable
without needing to set a token in the configuration file.

Closes #35693

Akasurde committed:
VMware: Documentation cleanup - vmware_guest (#35548)

Partially fixes: #35295

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

snecklifter committed:
Add GCE provider for ManageIQ (#35705)

This commit adds support for creating GCE providers on ManageIQ.

rcarrillocruz committed:
Remove unused enable_mode import on Junos cliconf plugin (#35722)

lberruti committed:
Fix: global name 'HAS_IPADDRESS' is not defined.

rcarrillocruz committed:
Document the multiline prompt/answer format for ios commands (#35716)

* Document the multiline prompt/answer format for ios commands

Fixes #34468

* Put a note

* Add also docs for vyos

trishnaguha committed:
fix eos_config second-level indent idempotece (#35588)

* fix eos_config second-level indent idempotece

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* address ganeshrn's comment - update doc

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

jcsmith committed:
Update example syntax in test_strategies.rst, (#35689)

jcsmith committed:
Update example syntax in playbooks_tests.rst. (#35688)

machacekondra committed:
Support High performance VM in oVirt (#35573)

* ovirt_vms: add cpu_mode

* ovirt_vms: add placement_policy

* ovirt_vms: add cpu_pinning

* ovirt_vms: add soundcard_enabled

* ovirt_vms: add smartcard_enabled

* ovirt_vms: add io_threads_enabled

* ovirt_vms: add ballooning_enabled

* ovirt_vms: add rng_device

* ovirt_vms: add custom_properties

* ovirt_vms: add memory_max

* ovirt_vms: Fix formatting

* ovirt_vms: add watchdog

sivel committed:
Tags on includes only affect the include. FIxes #35729 (#35732)

mattclay committed:
Extend httptester CA cert lifetime.

Akasurde committed:
VMware: Fix regression in vmware_portgroup (#35700)

This fix corrects the regression introduced by refactor commit in
vmware_portgroup module.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: vmware_guest_file_operation refactor (#35560)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: add new module vmware_host_firewall_facts (#35467)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: add new module : vcenter_folder (#35608)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

kandsten committed:
Detect OpenStack guests on RDO based installs, Newton and newer. (#35452)

The product string (used as the DMI product name) changed from
"OpenStack Nova" to "OpenStack Compute" between Mitaka and Newton
in the OpenStack RDO packaging.

maxamillion committed:
fix new EC2 C5 instance virtualization_type identification (#35063)

* fix new EC2 C5 instance virtualization_type identification

Fixes #35051

Signed-off-by: Adam Miller <admiller@redhat.com>

Akasurde committed:
VMware: add new module : vmware_host_lockdown (#35715)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sivel committed:
Match recent Dockerfile change in httptester playbook file, to extend ca cert lifetime (#35734)

Akasurde committed:
VMware: add new module vmware_host_dns_facts (#35708)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: add new module vmware_vmkernel_facts (#35535)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

abadger committed:
When tempdir fails to be created, we have nothng to remove

This commit should get us a better error traceback when things do go
wrong.

References #35613

Akasurde committed:
VMware: refactor vmware_cfg_backup (#35695)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

perlpunk committed:
Add examples to YAMLSyntax; fix minor issues and improve wording (#35703)

* Add examples to YAMLSyntax; fix minor issues and improve wording

I added proper titles to "Flow Collections" and "Block Scalars", because these
are official YAML Terms, so people can find better help when having issues.

I changed the "ignored newlines" to "folded newlines" because I think ignoring
gives a wrong impression, especially since empty lines (more than one newline)
are not ignored and have a special meaning.

I added an example how to enforce a newline in a folded block scalar.

I fixed the list of characters that are allowed/not allowed in plain and
double quoted scalars.

I added the #yaml channel to the IRC link.

I added links to the YAML Specs. For completeness, I also included YAML 1.2,
although PyYAML and libyaml currently implement 1.1.

* Improve wording

* Add three more characters to forbidden first characters

resmo committed:
exo_dns_record: remove limitation for multiple records only for A type (#35173)

ivanbojer committed:
new module panos_op.py (#34428)

* new module panos_op.py

* pep8 changes

Commit:
Fixes #35629 (#35668)

maxamillion commented:
Thanks for your submission! The code looks good, but I noticed there's no addition to the docs of the module that mention support of DragonflyBSD or any other considerations needed. Could you please add those as necessary? Thanks.

bcoca commented on lib/ansible/modules/system/user.py:
```diff
@@ -935,6 +935,207 @@ def modify_user(self):
         return (rc, out, err)
 
 
+class DragonFlyUser(User):
```
maybe just subclass freebsd one and then if there is a divergence you can override those methods

i would prefer that over soo much duplicate code

jcsmith committed:
Add examples for OpenBSD hosts. (#35691)

* Add examples for OpenBSD hosts.

* Incorporate changes recommended based on feedback from my pull request.

Commit:
[new module] azure_rm_keyvault (#35150)

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* updates to azure_rm_keyvault

* fix update bugs, add update test

samdoran committed:
Fix panos_object docs so they generate properly (#35756)

Change description field to a list rather than folder scalar.

Cleanup formatting and line breaks in README.

Akasurde committed:
VMware: Add new module : vmware_host_firewall_manager (#35596)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

dgoetz committed:
Add the option to use config file instead of enviroment variables (#32497)

and some error handling for the requests

bcoca committed:
add manpage install target

(cherry picked from commit 8e8a29f20a5b250947e00613033775ddd952430f)
(cherry picked from commit f572703efe65b8efedf3792d342ee3c09d9ba735)

sylr committed:
Fix typo in azure_rm_subnet (#35731)

Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

Akasurde committed:
FreeIPA: Add Env variable as module parameter (#35391)

This fix adds environment variable fallback method to read
argument parameters if user has not specified.

Fixes: #35368

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Fix ios enable password regex in terminal plugin (#35741)

Fixes #33993

On some ios devices the enable password prompt does not
end with a space, add a fix to handle this scenario.

mmercer-apixio committed:
Fixing syntax for mail callback doc (#35379)

mikewiebe committed:
Minor updates to nxos_nxapi tests (#35750)

* Minor updates to nxos_nxapi tests

* Fixed duplicate yaml key issue

* Make rubocop happy

mwperina committed:
Fix the oVirt high_performance VM type description.

ar7z1 committed:
influxdb_user: implement user password change (#35471)

* Added tests on user removing

* Implemented password changing

* Fix after review

* Added zhhuta changes

dagwieers committed:
Add initial ACI guide (#35364)

* Add initial ACI guide

* Implement all review suggestions

kwohlfahrt committed:
Add vfat support for the filesystem module (#23527)

* Add fat filesystem support

fatresize is temporarily disabled

* Refactor Filesystem.get_dev_size

For more sharing with vFAT class

* Fix filesystem tests on some OSs

I think this is due to older mke2fs on those systems.

* Fix vFAT command on FreeBSD

newfs doesn't seem to work on image files

* Refactor filesystem.grow()

Split out grow_cmd generation and Device operations

* Use swap as unsupported filesystem

Except FreeBSD, which doesn't have mkswap

* Be consistent about str(dev) vs dev.path

Prefer str(dev), this works transparently with '%s' formatting.

* Enable vfat resize, only test fatresize >= 1.0.4

Lower versions have a segfault bug.

* Only install fatresize where available

FreeBSD, OpenSUSE, RHEL and CentOS < 7 don't ship it.

Akasurde committed:
VMware: Detect SDRS for datastore cluster (#35644)

This fix adds a fallback mechanism where Storage Datacluster
is not enabled with SDRS. If user has Storage Datacluster without SDRS
then we will not get any datastore recommendation. This will fallback to
normal datastore selection method.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

brunocalogero committed:
adding aci_interface_policy_leaf_policy_group integration tests (#35486)

* adding aci_interface_policy_leaf_policy_group integration tests

* Added more ACI options for testing

brunocalogero committed:
added aci_switch_policy_vpc_protection_group module (#35769)

* added aci_switch_policy_vpc_protection_group module

* Small change wrt. new interface

rvgate committed:
There is no master branch in this project (#35773)

Since there is no master branch, best not to mention it in the issue
template to avoid confusion.

chifu1234 committed:
nxos_facts: svi support ipv4 (#35222)

* nxos_facts: svi support ipv4

* nxos_facts: copy paste error

* https://app.shippable.com/github/ansible/ansible/runs/51845/1/tests

* E111 indentation is not a multiple of four

* Update nxos_facts.py

Akasurde committed:
Remove get_exception from modules

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

nkakouros committed:
Fix config option descriptions (#35638)

* Fix config option descriptions

* Better describe ANSIBLE_PRIVATE_ROLE_VARS

giovannisciortino committed:
Fixes: #20986 blockinfile create undelying directory when create=true (#27961)

Commit:
Changing ENOS action plugin license from BSD to GPL (#34888)

* Changing BSD to GPL

* Updating License Information

* Updating Copyright year to 2017

* Changed copy right info

rambleraptor committed:
[cloud] New GCP module: DNS Managed Zones (gcp_dns_managed_zone.py) (#35014)

ryansb committed:
Add gcp_dns_managed_zone to CHANGELOG

newswangerd committed:
commented out check tests (#35788)

Commit:
Netscaler various fixes (#34800)

* Add default lb vserver option in netscaler_cs_vserver
Add documentation for ssl_certkey option in netscaler_cs_vserver

* Add options for gracefully disabling netscaler_server

* Add "state" suboption for netscaler_servicegroup servicemembers
Fix servicemember modification algorithm in netscaler_servicegroup
Fix monitorbindings modification algorithm in netscaler_servicegroup

IPvSean committed:
hostname parameter should be host_name (#35798)

the hostname parameter is wrong.  It will throw an error: "Unsupported parameters for (vyos_system) module: hostname Supported parameters include: domain_name, domain_search, host, host_name, name_server, password, port, provider, ssh_keyfile, state, timeout, username"  It is supposed to be host_name.  Only the example for documentation seems to be wrong in the module.

ivanbojer committed:
new panos_match_rule module (#35202)

* new panos_match_rule module

giovannisciortino committed:
lineinfile creating underlying directory manage permission denied exception (#35794)

ganeshrn committed:
Update network section in 2.5 porting guide (#35643)

* Update 2.5 porting guide with network module_utils move

Commit:
Apply regex to dir name only (#35744)

mattclay committed:
Temp install of setuptools for import coverage. (#35752)

aajdinov committed:
New module: manage 1&1 monitoring policy (cloud/oneandone/oneandone_monitoring_policy) (#35068)

* Added server_type parameter - defaults to K8S

* Changed server_type default value to cloud.

* Added server_type entry to documentation and updated server_type choice K8S to k8s_node

* Added support for overriding the default ONEANDONE_API_URL, and custom wait_timeout period for wait_for methods.

* Added firewall_policy, load_balancer, and monitoring_policy cloud modules for oneandone provider.

* Updated OneAndOneResources class and argument specs for monitoring policy and load balancer.

* Addressed ansible-test sanity pep8 and validate-modules errors.

* Removed extra modules to adhere to one module per PR.

* Fixed a typo

Yvan-Masson committed:
Fix typo (#35810)

jmcgill298 committed:
Pathatt mode (#35804)

* ACI_STATIC_BINDING: Update parameter to use more standard network lingo

* Add 'mode' as an alias

mikedlr committed:
Aws ssm multiple fixes (#35569)

* aws ssm parameter lookup test case - fails demonstrating no exception when parameter missing

* aws ssm parameter lookup - fail in case parameter doesn't exist

* aws ssm parameter lookup test case - failing case for nice return from path lookup

* aws ssm parameter lookup - convert incoming taglist to a key-value dictionary

* aws ssm parameter lookup - pep8 / style clean up

* aws_ssm lookup plugin rewrite for more standard interface

* aws_ssm module and lookup - introduce integration test and fix:

* aws_ssm module and lookup - error case integraton test and many PEP8 and other cleanups

* aws ssm parameter lookup - Various fixes in response to review + recursive fix & test

* aws ssm parameter lookup - more in response to review - shertel/abadger

* aws ssm parameter lookup unit test - move to mocker according to abadger

* aws ssm parameter lookup - integrate with new documentation fragment

* aws ssm parameter lookup - accept either aws_profile or boto_profile

* aws ssm parameter lookup - eliminate lookup document fragment until env vars are fixed later

abadger committed:
Enable wildcard import pylint (#35786)

* Wildcard imports should be taken care of.  Enable the pylint check for them
* Remove wildcard import code-smell test as we're now checking via pylint
* Add unused-wildcard-import as ignored in our compat code.

These three files use wildcard imports so that they can export
symbols in a compatible location.  The real code lives elsewhere.
So disable the pylint tests for the relevant sections of code.

jmcgill298 committed:
ACI_STATIC_BINDING: Update paths paramater (#35807)

* ACI_STATIC_BINDING: Update paths paramater

* Fix small issue

pilou- committed:
dimensiondata: fix broken import (#35634)

* Fix broken import

* Remove never reached statements

exit_json and fail_json methods call sys.exit, subsequent statements
can not be executed.

Commit:
fix nxos_vpc_interface peer_link idempotence (#35737)

* fix peer_link idempotence

* review comment

ianphil committed:
Az mod keyvault key (#33607)

* key module init

* Create/Delete pass sanity

* Integration tests passed

* Added check_mode

* Updated to support tags

* updated key tests

* fixed sanity

* added things that went missing during rebase

* fixed sanity

* fixed doc

* fix copyright

ianphil committed:
Az mod keyvault ops (#33606)

* keyvault_operations module init + template

* Created KeyVault Secrets Module. No Tests

* Passing 2.7/3.x sanity tests

* added intergration tests for keyvault secrets noop

* integration test pass/need keyvault module

* Added az env cleanup to intergration tests

* Updated auth to 0.3.6 ver

* fixed typo in integration tests

* Added check_mode

* Added tags

* Updated tags test, sanity + integration passed

* checked mode

* updated tests

* fixing sanity

* fix copyright

caphrim007 committed:
Fixes repeated calls to create an API object (#35816)

This patch fixes repeated attempts that the module would make to
re-create an API object. The change stores a copy for later lookup
instead. This prevents uncontrolled tokens from being created.

trishnaguha committed:
always remove interface after testing (#35822)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

jmcgill298 committed:
Pathatt connection (#35806)

* add interfacce_type to absent required if...not sure how this would work without it

* move fex requirement to required_if

* Fix incorrect documentation about port_channels

evrardjp committed:
Add etcd3 to botmeta (#35824)

The etcd3 module should have a corresponding entry in BOTMETA.

mkrizek committed:
Fix jinja2>=2.9 nested include vars (#35099)

This fixes a bug when parent's local vars where not available in nested
includes. The bug can only be seen with jinja>=2.9 which changes
how the variable scopes work.

Fixes #34886

jborean93 committed:
win_uri: fixes (#35487)

* win_uri: moved away from Invoke-WebRequest and fixed multiple issues

* fixes from review

mscherer committed:
Remove deadcode since diff is never used (#35147)

Flake8 keep warning me about that

resmo committed:
changelog: add new modules (#35712)

all new modules to 0c5650ee1bdf9dea75a235888c91b4cf8be90efd

willthames committed:
New aws_waf_web_acl module (#33212)

* Add aws_waf_web_acl rule

* Add new aws_waf_web_acl to CHANGELOG

wimnat committed:
[cloud] New AWS module - elb_target (#26483)

* New module - elb_target

* Work to add deregister functionality

* Add jurajseffer work on waiting for status change

* List the set of statuses as a choices list

* default target_status_timeout is 60

* Add 'unavailable' as a target status choice per API docs

* Add support to remove targets that are in unused state

* add support for availability zone selection

Fix deregistering targets from used target groups

Fix waiting for unused state for targets that have started deregistering

ryansb committed:
Add elb_target to CHANGELOG

tuxillo committed:
service: Add support for DragonFly BSD

For now create a subclass of the FreeBsd class

aajdinov committed:
New module: manage 1&1 firewall policy (cloud/oneandone/oneandone_firewall_policy) (#35066)

* Added firewall_policy, load_balancer, and monitoring_policy cloud modules for oneandone provider.

* Updated OneAndOneResources class and argument specs for monitoring policy and load balancer.

* Addressed ansible-test sanity pep8 and validate-modules errors.

* Removed extra modules to adhere to one module per PR.

aajdinov committed:
New module: manage 1&1 load balancer (cloud/oneandone/oneandone_load_balancer) (#35067)

* Added firewall_policy, load_balancer, and monitoring_policy cloud modules for oneandone provider.

* Updated OneAndOneResources class and argument specs for monitoring policy and load balancer.

* Addressed ansible-test sanity pep8 and validate-modules errors.

* Removed extra modules to adhere to one module per PR.

aajdinov committed:
New module: manage 1&1 private network (cloud/oneandone/oneandone_private_network) (#35076)

* Added firewall_policy, load_balancer, and monitoring_policy cloud modules for oneandone provider.

* Updated OneAndOneResources class and argument specs for monitoring policy and load balancer.

* Addressed ansible-test sanity pep8 and validate-modules errors.

* Added firewall_policy, load_balancer, and monitoring_policy cloud modules for oneandone provider.

* Updated OneAndOneResources class and argument specs for monitoring policy and load balancer.

* Addressed ansible-test sanity pep8 and validate-modules errors.

* Added oneandone_private_network cloud module for oneandone cloud provider. Removed extra modules to adhere to one module per PR.

* Updated check mode scenario for remove members, using proper wait_for method in remove_network, renamed private_network_id argument to private_network.

* Fixing validate-modules issue - Documented datacenter.

* Added oneandone_private_network cloud module for oneandone cloud provider. Removed extra modules to adhere to one module per PR.

Qalthos committed:
Update CHANGELOG.md

mkrizek committed:
redhat_subscription: do not use defaults from the conf file (#33763)

* redhat_subscription: do not read defaults from the rhsm.conf file, let subman provide those.

* Only pass options specified by user to the config

beardymcbeards committed:
[WIP] Add new network module for EdgeOS (#33109)

* Add edgeos network module using network_cli

* Add documentation and some sanity test fixes

* Update copyright and documentation

mkrizek committed:
Remove jinja native types from roadmap (#35444)

bcoca committed:
undeprecate include

defering deprecation until we are comfortable that the new actions
cover most of the ground users need.

bcoca committed:
remove runtime msg also

Commit:
Fix the example - a cluster is a required propery (#35836)

machacekondra committed:
Improve ovirt_vms, ovirt_hosts and ovirt_storage_domains (#35782)

* ovirt_vms: Add graphics consoles

* ovirt_hosts: Fix kdump integration check

* ovirt_storage_domains: Support multiple targets for block storages

* ovirt_vms: add cloud_init_persist parameter

* ovirt_vms: Change IO threads to int

* ovirt_vms: Fix update check

* ovirt_vms: Remove choices of OS, to not enforce it

machacekondra committed:
ovirt: Require Python SDK 4.2.4 for Ansible 2.5 (#35841)

ganeshrn committed:
Update changelog (#35848)

jontow committed:
support DragonFlyBSD by subclassing FreeBsdUser in modules/system/user.py

jontow commented:
botched rebase.. closing, will resubmit.
